### PR TITLE
build spinner component/service

### PR DESCRIPTION
**Pull request by vanesora**

### CODE REVIEW

File: src/app/pages/login/login.component.ts [hunk 1]
```diff
@@ -40,8 +43,10 @@ export class LoginComponent implements OnInit {
             role: 'admin' //change when we have user collection
           });
         this.router.navigate(['/home']);
+        this.spinnerService.close();
       }).catch(err => {
         this.errorlogin = true;
+        this.spinnerService.close();
       });
```
Comment (JuanSeBestia) [inline src/app/pages/login/login.component.ts:50]: Instead to put two times   `this.spinnerService.close();` use finally

Comment (JuanSeBestia) [reply src/app/pages/login/login.component.ts:50]:
```suggestion
      }).catch(err => {
        this.errorlogin = true;
      }).finally(()=>{  this.spinnerService.close();});
```

File: src/app/shared/components/header/header.component.ts [hunk 2]
```diff
@@ -7,9 +10,23 @@ import { Component, OnInit } from '@angular/core';
...
   }
 
+  logOut() {
+    this.spinnerService.openAlertDialog();
+    this.authService.logout().then(() => {
+      this.router.navigate(['/login']);
+      this.spinnerService.close();
+    }).catch(() => {
+      this.router.navigate(['/login']);
+      this.spinnerService.close();
```
Comment (JuanSeBestia) [inline src/app/shared/components/header/header.component.ts:29]: x2

File: src/app/shared/services/spinner.service.ts [hunk 3]
```diff
@@ -1,9 +1,29 @@
...
+  constructor(
+    private dialog: MatDialog,
+    private dialogRef: MatDialogRef<SpinnerComponent>
+  ) { }
 
-  constructor() { }
+  openAlertDialog() {
+    this.dialogRef = this.dialog.open(SpinnerComponent, {
+      data: {
+        message: 'HelloWorld',
```
Comment (KaterineMoon) [inline src/app/shared/services/spinner.service.ts:17]: El Hello World es mientras se termina el desarrollo?

File: src/app/pages/login/login.component.ts (hunk 1 above)
Comment (vanesora) [reply src/app/pages/login/login.component.ts]: done

File: src/app/shared/components/header/header.component.ts (hunk 2 above)
Comment (vanesora) [reply src/app/shared/components/header/header.component.ts:29]: done

File: src/app/shared/services/spinner.service.ts (hunk 3 above)
Comment (vanesora) [reply src/app/shared/services/spinner.service.ts]: nop, ya lo quité

**Pull request merged**